File: bitget-node-sdk-api/src/lib/v3/UtaLoanApi.ts
import {BaseApi} from '../BaseApi';

export class UtaLoanApi extends BaseApi {

    // GET /api/v3/loan/coins
    coinInfos(qsOrBody: object) {
        const url = '/api/v3/loan/coins';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/loan/interest
    interests(qsOrBody: object) {
        const url = '/api/v3/loan/interest';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/loan/borrow
    borrow(qsOrBody: object) {
        const url = '/api/v3/loan/borrow';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/loan/borrow-ongoing
    loanOrders(qsOrBody: object) {
        const url = '/api/v3/loan/borrow-ongoing';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/loan/borrow-history
    borrowHistory(qsOrBody: object) {
        const url = '/api/v3/loan/borrow-history';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/loan/repay
    repay(qsOrBody: object) {
        const url = '/api/v3/loan/repay';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/loan/repay-history
    repayHistory(qsOrBody: object) {
        const url = '/api/v3/loan/repay-history';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/loan/revise-pledge
    revisePledge(qsOrBody: object) {
        const url = '/api/v3/loan/revise-pledge';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/loan/pledge-rate-history
    pledgeRateHistory(qsOrBody: object) {
        const url = '/api/v3/loan/pledge-rate-history';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/loan/debts
    debts(qsOrBody: object) {
        const url = '/api/v3/loan/debts';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/loan/reduces
    reduces(qsOrBody: object) {
        const url = '/api/v3/loan/reduces';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }
}

File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v3/UtaLoanApi.java
package com.bitget.openapi.api.v3;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface UtaLoanApi {

    @GET("/api/v3/loan/coins")
    Call<ResponseResult> coins(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/loan/interest")
    Call<ResponseResult> interest(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/loan/borrow")
    Call<ResponseResult> borrow(@Body Map<String, Object> paramMap);

    @GET("/api/v3/loan/borrow-ongoing")
    Call<ResponseResult> borrowOngoing(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/loan/borrow-history")
    Call<ResponseResult> borrowHistory(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/loan/repay")
    Call<ResponseResult> repay(@Body Map<String, Object> paramMap);

    @GET("/api/v3/loan/repay-history")
    Call<ResponseResult> repayHistory(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/loan/revise-pledge")
    Call<ResponseResult> revisePledge(@Body Map<String, Object> paramMap);

    @GET("/api/v3/loan/pledge-rate-history")
    Call<ResponseResult> pledgeRateHistory(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/loan/debts")
    Call<ResponseResult> debts();

    @GET("/api/v3/loan/reduces")
    Call<ResponseResult> reduces(@QueryMap Map<String, String> paramMap);
}
